tests/testthat/test-hypothesis_test.R

```diff
@@ -151,6 +158,7 @@ if (suppressWarnings(requiet("testthat") && requiet("ggeffects") && requiet("mar
     })
     test_that("hypothesis_test, categorical, NULL", {
       out <- hypothesis_test(model3, c("groups", "episode"), test = NULL)
+      out <- out[order(out$groups, out$episode),]
```

Comment:
If this is the major change you made, can't we just "revert" the sorting in `hypothesis_test()`, so actually nothing changes here?

Reply:
In general, I don't like sorting rows *after* the fact for `marginaleffects` output, because it could break the alignment between bootstrap or bayesian draws and the printed output. But that may be irrelevant for your package; I'm not sure exactly what feature set you support.

Reply:
> because it could break the alignment between bootstrap or bayesian draws and the printed output

In what way? Do you have an example? `hypothesis_test()` is mainly about two things: bringing the data (data grid) in shape for the call to `predictions()` etc., and then especially manipulating the returned data frame for printing.

Reply:
``` r
library(brms)
library(dplyr)
library(marginaleffects)
mod <- brm(mpg ~ qsec + cyl, data = mtcars)

# sorted output
cmp <- avg_comparisons(mod) |> arrange(desc(term))

# draws and rows are no longer properly aligned
cmp |> posterior_draws(shape = "PxD") |> apply(1, median)
#         b1         b2 
# -2.9821892 -0.6239233

cmp$estimate
# [1] -0.6239233 -2.9821892
```

Reply:
ah, ok. That only applies if you want to bind results from _marginaleffects_ and _brms_ together? I don't think this is a scenario I have in mind for `hypothesis_test()`.